src/iris/cache.py

```diff
@@ -3,6 +3,7 @@
 
 from . import db
 import logging
+import threading
```

Comment:
We should use gevent.Lock instead of threading.Lock

Reply:
does it matter though since we're monkey patching?

Reply:
Sorry, I made a typo: we should use `gevent.lock.Semaphore`. 

It looks like gevent does monkey patch the threading module per `https://www.gevent.org/api/gevent.monkey.html#gevent.monkey.patch_thread` but when we interact with these primitives directly it may be more clear/explicit to use the gevent equivalents, and not rely on the monkey patch ordering to fix this for us (as afaik in our internal uwsgi wrapper we only patch ssl to start with)

This would also be consistent with how we're doing locking in the rate limiting module: https://github.com/linkedin/iris/blob/master/src/iris/sender/quota.py

Reply:
like James said it gets monkey patched but it does seem clearer and more consistent to use the gevent lock explicitly. Plus we can avoid any weird bugs in the future.